06-building-a-program-start-to-finish/writing-tests/output/spec/InvertedIndex.spec.js

```diff
@@ -0,0 +1,279 @@
+      expect(indexInstance.searchIndex(term, 'books'))
+      .toBeFalsy();
+    });
+    it('should return an empty object for words not found', () => {
+      const term = 'Aeroplane';
+      const expectedOutput = indexInstance.searchIndex(term, 'books');
+      expect(expectedOutput[0].indexes).toEqual({ });
+    });
+    it('should return appropriate result for when all files is selected',
+    () => {
+      const expectedOutput =
+        [{ indexes: { alice: [0], the: [1, 2] },
+          searchedFile: 'books',
+          documents: [0, 1, 2] },
+        { indexes: { alice: [0],
+          is: [0],
+          the: [0, 1] },
+          searchedFile: 'anotherBook',
+          documents: [0, 1] },
+        { indexes: { },
+          searchedFile: 'term',
+          documents: [] }];
+      expect(indexInstance
+        .searchIndex('Alice is the', 'all')).toEqual(expectedOutput);
+    });
+  });
+
+  describe('Tokenize words', () => {
+    it('should strip out special characters from text in documents', () => {
+      let excerpt = 'Alice l##$oves her ima&&gination?';
+      const expectedTokens = ['alice', 'loves', 'her', 'imagination'];
+      excerpt = InvertedIndex.tokenize(excerpt);
+      expect(excerpt).toEqual(expectedTokens);
+    });
+  });
+
+  describe('Get index', () => {
+    it('should return the appropriate output for the given filename', () => {
+      const filename = 'books';
+      const expectedOutput = { alice: [0],
+        falls: [0],
+        into: [0],
+        a: [0, 1, 2],
+        rabbit: [0],
+        hole: [0],
+        and: [0, 1, 2],
+        enters: [0],
+        world: [0],
+        full: [0],
+        of: [0, 1, 2],
+        imagination: [0],
+        in: [0],
+        wonderland: [0],
+        an: [1, 2],
+        unusual: [1, 2],
+        alliance: [1, 2],
+        man: [1, 2],
+        elf: [1, 2],
+        dwarf: [1, 2],
+        wizard: [1, 2],
+        hobbit: [1, 2],
+        seek: [1, 2],
+        to: [1, 2],
+        destroy: [1, 2],
+        powerful: [1, 2],
+        ring: [1, 2],
+        the: [1, 2],
+        lord: [1, 2],
+        rings: [1, 2],
+        fellowship: [1, 2] };
+      expect(indexInstance.getIndex(filename))
+        .toEqual(expectedOutput);
+    });
+    it('should return false for an empty filename', () => {
+      const filename = '';
+      expect(indexInstance.getIndex(filename))
+        .toBeFalsy();
+    });
+  });
+
+  describe('Validate File', () => {
+    it('should return false for incorrect document structure', () => {
+      const term = [{ t1: 'Welcome home', text: 'This is really home' }];
+      expect(InvertedIndex.validateFile(term, 'term')).toBeFalsy();
+    });
+    it('should return true for correct document structure', () => {
+      const term = [{ title: 'Welcome home', text: 'This is really home' }];
+      expect(InvertedIndex.validateFile(term, 'term')).toBeTruthy();
+    });
+  });
+
+  describe('Get Document Tokens Data', () => {
```

Comment:
I believe what you are testing is the _method_ you have in your InvertedIndex and not the _method **data**_. Please correct this language across your test suites.